convex_hull.cpp

```diff
@@ -158,13 +158,38 @@ class convex_hull : public CDLlist< point< T > >
 };
 
 int main() {
+
+	freopen("input.txt", "r", stdin);
+	freopen("output.txt", "w", stdout);
+
 	convex_hull<double> list;
-	point<double> p1(1, 2), p2(2, 3), p3(3, 1), p4(4, 4), p5(3, 2);
-	list.insert(p1);
-	list.insert(p2);
-	list.insert(p3);
-	list.insert(p4);
-	list.insert(p5);
+	
+	long long int cases;
+	if(scanf("%lld",&cases) != 1)
```

Comment:
We don't need `cases`. If `scanf` returns `-1` then I think the file has nothing to read.

Reply:
yes, this is true, `scanf` will return `-1`, in case when there is nothing in file, but it would be better if User also gives how many values(points) will it give, it is just me being too much used to input format of CP contests.

Reply:
Rename `cases` to something that fits more in the context of the program.